### DISCUSSION TITLE
Timezone as text

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2842] [Q&A] [answered]
In a postgres database I get this error:

```
column "my_date" is of type timestamp with time zone but expression is of 
type text hint: You will need to rewrite or cast the expression.
```

in my build.yaml I use `store_date_time_values_as_text: true` and this is the format of the datime (timestampz type) in my database `2024-01-08 10:51:12+00` what should/can I do?

**Comment by simolus3** [comment, marked as answer]
Drift's `dateTime()` column can't be used with postgres dates - it always expects a text/int column because that's all we get with sqlite3.

With postgres, you can use

```dart
  TimestampColumn get myDate => customType(PgTypes.timestampNoTimezone)();
```

In the next version, `PgTypes.timestampWithTimezone` will be available as well. But as long as you're connecting to an existing database (that is, you're not letting drift emit the `CREATE TABLE` statements), the two time types are identical.

**Reply by BananaMasterz** [reply]
Amazing as always, thanks!
